### ISSUE TITLE
Following redirects when POSTing a file

### ISSUE THREAD

**Issue by za3k** [rest-client/rest-client#244]
While trying to follow a redirect, I noticed that an included file (as per a form) does not seem to work, and instead I get a message about a closed error stream.  I assume this is because of the recent switch treating files as streams, which wouldn't be able to send the file twice, but there doesn't seem to be a generic way around it this Ruby novice can find.

I'm not looking for support as I've dealt with the root problem in my case. I'm just filing a (quite likely mistaken) bug about a regression in redirection capabilities.

```
filename = "IntroCyb.pdf"
uploaded_file = site['StorageService'].post(:file => File.new(filename, 'rb')) do |response, request, result, &block|
    if [301, 302, 307].include? response.code
        response.follow_redirection(request, result, &block) # does not work for file stream, actually
    else
        response.return!(request, result, &block)
    end
end
```

**Comment by soheilbm** [comment]
Hi @vanceza, I'm also facing the same issue so I was wondering did you figure out how to fix this?

**Comment by vanceza** [comment]
No, it's not fixable without major rework, I just did something like "use the redirected URL to begin with"
